pkg/cloudprovider/pricing_test.go

```diff
@@ -199,7 +199,7 @@ var _ = Describe("Pricing", func() {
 			},
 		})
 		p := NewPricingProvider(ctx, fakePricingAPI, fakeEC2API, "", false, make(chan struct{}))
-		Eventually(func() bool { return p.OnDemandLastUpdated().After(updateStart) }, 5*time.Second).Should(BeTrue())
+		Eventually(func() bool { return p.SpotLastUpdated().After(updateStart) }, 5*time.Second).Should(BeTrue())
```

Comment:
Don't know if this matters, but do we also need to change the values of the fake.NewOnDemandPrice() to use fake.NewSpotPrice()? 

We hard code on-demand here, so I don't know if that changes anything for lines 
103, 104, 139, 140, 168, 169, 197, 198?
 https://github.com/aws/karpenter/blob/main/pkg/fake/pricingapi.go#L61